### PR TITLE
pythonPackages.cfn-lip: init at 1.1.0

### PR DESCRIPTION
**Pull request by PsyanticY**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -2043,6 +2043,8 @@ with pkgs;
 
   certstrap = callPackage ../tools/security/certstrap { };
 
+  cfn_flip = callPackage ../development/tools/cfn_flip { };
```
Comment (dotlambda) [inline pkgs/top-level/all-packages.nix]:
```suggestion
  cfn-flip = callPackage ../development/tools/admin/cfn-flip { };
```

File: pkgs/development/tools/cfn_flip/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ lib, python3Packages }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "cfn_flip";
```
Comment (dotlambda) [inline pkgs/development/tools/cfn_flip/default.nix]:
```suggestion
  pname = "cfn-flip";
```
You will need to specify `pname = "dfn_flip"` inside `fetchPypi`.

File: pkgs/development/tools/cfn_flip/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  version = "1.1.0";
```
Comment (dotlambda) [inline pkgs/development/tools/cfn_flip/default.nix]:
```suggestion
  version = "1.1.0.post1";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/cfn_flip/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "11fn4capv4137qsq8lgi1dmx6q5f5q83mlrhjhsvi6pkn3nibynk";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ six pyyaml click tox pytestrunner ];
```
Comment (dotlambda) [inline pkgs/development/tools/cfn_flip/default.nix]:
```suggestion
  propagatedBuildInputs = with python3Packages; [ six pyyaml click ];
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by PsyanticY** [comment]
@dotlambda  Thanks for the review. addressed requested changes, 
for `pytestunner` it is required in the [package `setup.py`](https://github.com/awslabs/aws-cfn-template-flip/blob/1ebb4da20f1e256397294abd4d0e9620ccbbce57/setup.py#L29)

**Comment by dotlambda** [comment]
> for pytestunner it is required in the package setup.py

Then it should not be propagated. @FRidh I'm still not whether to use `buildInputs` or `propagatedBuildInputs` in that case.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Judging from their [README](https://github.com/awslabs/aws-cfn-template-flip#python-package) it looks this should be moved into `python-packages.nix`. Shouldn't it?

**Comment by PsyanticY** [comment]
@dotlambda  I think `nativeBuildInputs` would do the job ?
I think it should not be in `python-packages.nix`,  since most of the people would use it as a binary to convert form Json to yaml and not  as a package in other python code. But, it is up to you and @FRidh to decide.

**Comment by dotlambda** [comment]
There's no harm in putting it in `python-packages.nix` and since it can be imported as a Python library it should be put there.

**Comment by PsyanticY** [comment]
@dotlambda Moved the package to python-packages.

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -303,6 +303,8 @@ in {
   cleo = callPackage ../development/python-modules/cleo { };
 
   clikit = callPackage ../development/python-modules/clikit { };
+  
+  cfn_flip = callPackage ../development/python-modules/cfn_flip { };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: f comes before l if I remember correctly

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please set the commit message to this PR's title.

File: pkgs/top-level/python-packages.nix (hunk 5 above)
Comment (PsyanticY) [reply pkgs/top-level/python-packages.nix]: yeah right hhhhhhh @dotlambda

**Comment by PsyanticY** [comment]
@dotlambda fixed that ^^

File: pkgs/top-level/python-packages.nix (hunk 5 above)
Comment (dotlambda) [reply pkgs/top-level/python-packages.nix]: still wrong, as is the commit message
and I forgot to tell you to normalize attribute and filename to `cfn-flip`.

Comment (PsyanticY) [reply pkgs/top-level/python-packages.nix]: yeah seems like i didn't `git add .`

**Comment by PsyanticY** [comment]
@dotlambda  should be good now.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.cfn-flip python3.pkgs.cfn-flip

**Pull request merged**